src/native/java_class_ids.c

```diff
@@ -2065,6 +2081,28 @@ static void s_cache_mqtt5_on_disconnection_return(JNIEnv *env) {
     AWS_FATAL_ASSERT(mqtt5_on_disconnection_return_properties.return_constructor_id);
 }
 
+struct java_aws_mqtt5_listener_options_properties mqtt5_listener_options_properties;
+
+static void s_cache_mqtt5_listener_options(JNIEnv *env) {
```

Comment:
Interesting, it seems that we have been use getter functions for access the field id for all options right now. Is there any reason why we always use GetFieldID?

Reply:
You still need to use `GetFieldID`, but you can avoid calling the getter functions in Java and just access the properties directly.

Example:
* https://github.com/awslabs/aws-crt-java/blob/main/src/native/java_class_ids.c#L1551
* https://github.com/awslabs/aws-crt-java/blob/main/src/native/mqtt5_client.c#L2677

That said though, the only real benefit (that I know of) is that it allows us to have Java classes that contain properties/fields that are entirely private and not accessible to the customer using Java code, but we can still populate and modify these values in C as needed. This is also doable with private getter functions in Java as well, but at that point it is generally (though not always) easier to just get the field directly.

Regardless though, either way works totally fine and I don't think there is any big pro/con to either solution - more just a note that it can be done either way.

Reply:
I see. That's good to know. Thank you!